File: zonegen/txt_packing.py
from .constants import TXT_STRING_MAX_LEN


def pack_txt_strings(payload: str, max_len: int = TXT_STRING_MAX_LEN) -> list[str]:
    if not payload:
        return [payload]
    return [payload[i : i + max_len] for i in range(0, len(payload), max_len)]


def unpack_txt_strings(strings: list[str]) -> str:
    return "".join(strings)

File: webapp/txtPacking.js
// Mirrors zonegen/txt_packing.py::unpack_txt_strings -- plain concatenation of the
// already-separated RFC 1035 character-strings that make up one record's payload.
// Unwrapping Cloudflare's own quoting/spacing convention for multi-string TXT answers
// happens in doh.js, upstream of this -- by the time strings reach here they're already
// a clean list of the original packed segments, same shape dnspython hands the Python
// downloader (downloader/dns_store.py).
export function unpackTxtStrings(strings) {
  return strings.join('');
}
